exoscale/api/compute.py

```diff
@@ -1688,37 +1682,35 @@ def create_elastic_ip(
         except CloudStackApiException as e:
             raise APIException(e.error["errortext"], e.error)
 
-        return ElasticIP._from_cs(self, res["ipaddress"])
+        return ElasticIP._from_cs(self, res["ipaddress"], zone=zone)
 
-    def list_elastic_ips(self, zone=None, **kwargs):
+    def list_elastic_ips(self, zone, **kwargs):
         """
         List Elastic IPs.
 
         Parameters:
-            zone (Zone): a zone to restrict results to
+            zone (Zone): the zone to list in
             
         Yields:
             ElasticIP: the next Elastic IP
         """
 
         try:
             _list = self.cs.listPublicIpAddresses(
-                fetch_list=True,
-                zoneid=getattr(zone, "id", None),
-                iselastic=True,
-                **kwargs,
+                fetch_list=True, zoneid=zone.id, iselastic=True, **kwargs
             )
 
             for i in _list:
-                yield ElasticIP._from_cs(self, i)
+                yield ElasticIP._from_cs(self, i, zone=zone)
         except CloudStackApiException as e:
             raise APIException(e.error["errortext"], e.error)
 
-    def get_elastic_ip(self, address=None, id=None):
+    def get_elastic_ip(self, zone, address=None, id=None):
```

Comment:
I'm not sure why we need zone here, if retrieving a single IP we'd use the address or the id? (or is it future proof for when we have CS deployed across multiple zones?)